mautic/mautic
# Implementing DI for commands that still use container

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [n]
| BC breaks? (use the c.x branch)        | [y]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | Fixes https://mautic.atlassian.net/browse/TPROD-285



#### Description:



#### Steps to test this PR:

Several commands were refactored to use dependency injection instead of loading their dependencies from the container which is deprecated in Symfony 5. Run the commands bellow to check they work the same as before.

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. `bin/console mautic:cache:clear`
3. `bin/console mautic:update:apply`
4. `bin/console mautic:maintenance:cleanup`
5. `bin/console mautic:theme:json-config`
6. `bin/console mautic:update:find`
7. `bin/console mautic:assets:generate`
8. `bin/console mautic:install:data`
9. `bin/console mautic:campaigns:summarize`
10. `bin/console mautic:campaigns:trigger`
11. `bin/console mautic:campaigns:rebuild`
12. `bin/console mautic:messages:send`
13. `bin/console mautic:broadcasts:send`
14. `bin/console mautic:emails:send`
15. `bin/console mautic:contacts:deduplicate`
16. `bin/console mautic:segments:update`
17. `bin/console mautic:citrix:sync`
18. `bin/console mautic:transifex:pull`
19. `bin/console mautic:transifex:push`
20. `bin/console mautic:unusedip:delete`
21. `bin/console mautic:iplookup:download`
22. `bin/console mautic:install`
23. `bin/console mautic:integrations:sync`
24. `bin/console mautic:import`
25. `bin/console mautic:custom-field:create-column`
26. `bin/console mautic:integration:fetchleads`
27. `bin/console mautic:integration:pushleadactivity`
28. `bin/console mautic:queue:process`
29. `bin/console mautic:webhooks:process`
30. `bin/console mautic:integration:pipedrive:fetch`
31. `bin/console mautic:integration:pipedrive:push`

### BC Breaks
- Command `Mautic\LeadBundle\Command\CheckQueryBuildersCommand` and the methods it use:
- `Mautic\LeadBundle\Model\ListModel::getVersionNew()`
- `Mautic\LeadBundle\Model\ListModel::getVersionOld()`

Reason for removing it is that this command was used to compare different segments on the new segment query builder and the old one. But since the old one was removed since then it compares the new one with the new one so do not bring any value.

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11041"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
